transformations/slangificator/transformation.py

```diff
@@ -0,0 +1,217 @@
+                #Check if word is in the corresponding dictionary
+                if token.lemma_ in Slang_Adverbs[0]: 
+                    ReplPot+=1 # increment potential replacements
+        
+                    repDecision=random.uniform(0, 1)<=probReplaceAdverb # Randomly decide whether to replace or not
+                        
+                    if repDecision: #if replcement is made
+                        ReplMade+=1 
+        
+                        #Choose a new word for replacement                
+                        #ind=Slang_Adverbs[0].index(token.lemma_)
+                        indAll = [i for i, x in enumerate(Slang_Adverbs[0]) if x == token.lemma_] # all possible replacements 
+                        ind_r=random.randint(0, len(indAll)-1) # choose one of the replacements
+                        ind=indAll[ind_r] # index of that replacement                
+                        
+                        
+                        temp=Slang_Adverbs[1][ind] # pick the word chosen for replcacement
+                        if isCap: # Make the fist letter capital if necessary
+                            temp=temp[0].upper() + temp[1:] 
+                        modified_toks.append(temp + token.whitespace_)    
+                    
+                    else: #if no replcement is made
+                          modified_toks.append(token.text + token.whitespace_)            
+                    
+                else: # if not in the dictionary
+                    modified_toks.append(token.text + token.whitespace_)
+                    
+            #Adjectives
+            elif token.tag_ == 'JJ':
+                #Check if word is in the corresponding dictionary
+                if token.lemma_ in Slang_Adjectives[0]:      
+                    ReplPot+=1 # increment potential replacements
+        
+                    repDecision=random.uniform(0, 1)<=probReplaceAdjective # Randomly decide whether to replace or not
+                        
+                    if repDecision: #if replcement is made
+                        ReplMade+=1 
+        
+                        #Choose a new word for replacement                
+                        #ind=Slang_Adjectives[0].index(token.lemma_)
+                        indAll = [i for i, x in enumerate(Slang_Adjectives[0]) if x == token.lemma_] # all possible replacements 
+                        ind_r=random.randint(0, len(indAll)-1) # choose one of the replacements
+                        ind=indAll[ind_r] # index of that replacement                 
+                        
+                        temp=Slang_Adjectives[1][ind] # pick the word chosen for replcacement
+                        if isCap: # Make the fist letter capital if necessary
+                            temp=temp[0].upper() + temp[1:] 
+                        modified_toks.append(temp + token.whitespace_)   
+        
+                    else: #if no replcement is made
```

Comment:
The following is repeated three times which means it can probably be rewritten.

Reply:
I was thinking about it for a while but since I have had somewhat limited exposure to Python this way seems to be the most clear one from the logic point of view though it might not be the most "pythonic" one.

Reply:
@sebastianGehrmann, I think thanks to the suggestion from @kaustubhdhole I got an idea on how to rewrite the code to avoid logic duplication.